src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/ViewContext.cs

```diff
@@ -58,24 +59,24 @@ public ViewContext()
                 throw new ArgumentNullException(nameof(view));
             }
 
-            if (viewData == null)
+            if (writer == null)
             {
-                throw new ArgumentNullException(nameof(viewData));
+                throw new ArgumentNullException(nameof(writer));
             }
 
-            if (tempData == null)
+            if (htmlHelperOptions == null)
             {
-                throw new ArgumentNullException(nameof(tempData));
+                throw new ArgumentNullException(nameof(htmlHelperOptions));
             }
 
-            if (writer == null)
+            if (viewData == null)
             {
-                throw new ArgumentNullException(nameof(writer));
+                throw new ArgumentNullException(nameof(viewData));
```

Comment:
This is the constructor that `ViewComponentResult` and `ViewExecutor` use but `ViewExecutor` only provides a fallback `ITempDataDictionary` instance. Can't we still hit this `Exception` if `ViewData` is left `null` in a manually-crafted `ViewResult`?